### PR TITLE
Fixes some decoding troubles

### PR DESCRIPTION
**Pull request by lars18th**
This patch implements fixes described in:
- https://github.com/catalinii/minisatip/issues/547#issue-414611044
- https://github.com/catalinii/minisatip/issues/544#issuecomment-467194984

So, it improves the decoding.

### CODE REVIEW

File: src/stream.c [hunk 1]
```diff
@@ -1083,7 +1083,7 @@ int read_dmx(sockets *s)
 	if (!send)
 		return 0;
 
-	ad->flush = 0;
+	ad->flush = flush_all;
```
Comment (catalinii) [inline src/stream.c]: I think flush_all is not needed at all by looking at the code.

Comment (lars18th) [reply src/stream.c]: Ok. So perhaps it's a good idea to remove all code with `flush_all`.

File: src/dvbapi.c [hunk 2]
```diff
@@ -220,6 +220,7 @@ int dvbapi_reply(sockets *s)
 						k->filter_id[fpos] = -1;
 						k->pid[fpos] = -1;
 					}
+					break;
```
Comment (lars18th) [inline src/dvbapi.c]: Hi @catalinii ,

I'm sure this change is necessary.
It solves problems with my Oscam server.

File: src/stream.c (hunk 1 above)
Comment (catalinii) [reply src/stream.c]: Yes is better

File: src/dvbapi.c (hunk 2 above)
Comment (lars18th) [reply src/dvbapi.c]: Hi @catalinii ,

You tested this? It _really_ resolves troubles!

**Comment by catalinii** [comment]
Hi @lars18th,

I can accept this as long as you remove the flush_all code, or do not touch stream.c at all.

**Comment by lars18th** [comment]
> Hi @lars18th,
> 
> I can accept this as long as you remove the flush_all code, or do not touch stream.c at all.

Done! :wink:

File: src/pmt.c [hunk 3]
```diff
@@ -503,6 +503,8 @@ void update_cw(SPMT *pmt)
 		return;
 	}
 	LOGM("%s: pmt %d, parity %d, CW %d", __FUNCTION__, pmt->id, pmt->parity, pmt->cw ? pmt->cw->id : -1);
+	if (pmt->cw)
+		return;
```
Comment (lars18th) [inline src/pmt.c]: Hi @catalinii ,

This change fixes the trouble commented at #570

**Pull request merged**